components/org.wso2.carbon.identity.captcha/src/main/java/org/wso2/carbon/identity/captcha/util/CaptchaUtil.java

```diff
@@ -258,10 +259,29 @@ public static boolean isValidCaptcha(String reCaptchaResponse) throws CaptchaExc
         try {
             try (InputStream in = entity.getContent()) {
                 JsonObject verificationResponse = new JsonParser().parse(IOUtils.toString(in)).getAsJsonObject();
-                if (verificationResponse == null || verificationResponse.get("success") == null ||
-                        !verificationResponse.get("success").getAsBoolean()) {
+                if (verificationResponse == null) {
                     throw new CaptchaClientException("reCaptcha verification failed. Please try again.");
                 }
+                boolean success = verificationResponse.get(CaptchaConstants.CAPTCHA_SUCCESS) != null
+                        && verificationResponse.get(CaptchaConstants.CAPTCHA_SUCCESS).getAsBoolean();
+                if (verificationResponse.get(CaptchaConstants.CAPTCHA_SCORE) != null) {
+                    double score =  verificationResponse.get(CaptchaConstants.CAPTCHA_SCORE).getAsDouble();
+                    // reCAPTCHA v3 response contains score
+                    if (log.isDebugEnabled()) {
+                        log.debug("reCAPTCHA v3 success:" + success + ", action:" +
+                                verificationResponse.get("action") + ", score:" + score);
+                    }
+                    if (!success ||  score < scoreThreshold) {
```

Comment:
Shall we throw different messages for each case? To identify the verification failed because of not having success or the score is less than the threshold.